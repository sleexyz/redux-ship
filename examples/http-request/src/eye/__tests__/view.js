// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Eye from '../view';

function dispatch() {}

const states = [{
  color: null,
  isLoading: false,
}, {
  color: null,
  isLoading: true,
}, {
  color: 'red',
  isLoading: false,
}, {
  color: 'red',
  isLoading: true,
}];

test('view', () => {
  states.forEach(state => {
    const tree = renderer.create(<Eye dispatch={dispatch} state={state} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
