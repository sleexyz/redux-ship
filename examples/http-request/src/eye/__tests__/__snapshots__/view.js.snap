exports[`test view 1`] = `
<div>
  <button
    disabled={false}
    onClick={[Function]}>
    Get color
  </button>
</div>
`;

exports[`test view 2`] = `
<div>
  <button
    disabled={true}
    onClick={[Function]}>
    Loading...
  </button>
</div>
`;

exports[`test view 3`] = `
<div>
  <p>
    red
  </p>
  <button
    disabled={false}
    onClick={[Function]}>
    Get color
  </button>
</div>
`;

exports[`test view 4`] = `
<div>
  <p>
    red
  </p>
  <button
    disabled={true}
    onClick={[Function]}>
    Loading...
  </button>
</div>
`;
